core/src/main/scala/org/apache/spark/status/api/v1/ApplicationListResource.scala

```diff
@@ -42,7 +44,7 @@ private[v1] class ApplicationListResource(uiRoot: UIRoot) {
     }
     val includeCompleted = adjStatus.contains(ApplicationStatus.COMPLETED)
     val includeRunning = adjStatus.contains(ApplicationStatus.RUNNING)
-    val appList = allApps.filter { app =>
+    allApps.filter { app =>
```

Comment:
Yup I missed that part. Very good suggestion.

Now I've made changes to return a view of the list so that we can filter at first and then take the limit. The previous getApplicationInfoList remains unchanged. 

Thoughts?